apis/v1beta1/validation/gateway_test.go

```diff
@@ -247,6 +247,63 @@ func TestValidateGateway(t *testing.T) {
 			},
 			expectErrsOnFields: nil,
 		},
+		"ip address and hostname in addresses are valid": {
+			mutate: func(gw *gatewayv1b1.Gateway) {
+				gw.Spec.Addresses = []gatewayv1b1.GatewayAddress{
+					{
+						Type:  ptrTo(gatewayv1b1.IPAddressType),
+						Value: "1.2.3.4",
+					},
+					{
+						Type:  ptrTo(gatewayv1b1.HostnameAddressType),
+						Value: "foo.bar",
+					},
```

Comment:
Just for the sake of completion, we should add a valid ipv6 address to this test.

Reply:
added